lib/mshoplib/src/MShop/Coupon/Provider/FreeShipping.php

```diff
@@ -81,7 +81,9 @@ public function update( \Aimeos\MShop\Order\Item\Base\Iface $base ) : \Aimeos\MS
 		foreach( $base->getService( \Aimeos\MShop\Order\Item\Base\Service\Base::TYPE_DELIVERY ) as $service )
 		{
 			$price = $price->setRebate( $price->getRebate() + $service->getPrice()->getCosts() )
-				->setCosts( $price->getCosts() - $service->getPrice()->getCosts() );
+				->setCosts( $price->getCosts() - $service->getPrice()->getCosts() )
+				->setTaxRates( $service->getPrice()->getTaxRates() )
+				->setModified();
```

Comment:
`setModified()` should not be necessary because `setCosts()` and `setTaxrates()` will set the modified flag automatically.